### PR TITLE
feat: BuiltinScalarFunction::Cbrt

### PR DESCRIPTION
**Pull request by izveigor**
# Which issue does this PR close?
Part of https://github.com/apache/arrow-datafusion/issues/5838

# Rationale for this change

# What changes are included in this PR?

# Are these changes tested?
Yes

# Are there any user-facing changes?
Yes

### CODE REVIEW

**Review by yjshen — COMMENTED** [review]
Thanks @izveigor for the contribution! Could you please also update the documentation in both `docs/source/user-guide/expression.md` and `docs/source/user-guide/sql/scala_functions.md`?

File: datafusion/core/tests/sql/expr.rs [hunk 1]
```diff
@@ -1629,6 +1630,19 @@ async fn csv_query_sqrt_sqrt() -> Result<()> {
     Ok(())
 }
 
+// this query used to deadlock due to the call udf(udf())
```
Comment (alamb) [inline datafusion/core/tests/sql/expr.rs:1633]: I don't understand this comment

Comment (alamb) [reply datafusion/core/tests/sql/expr.rs:1633]:
```suggestion
```

Perhaps it was a left over from a copy / paste?

What do you think @izveigor ?

**Comment by alamb** [comment]
Thanks again @izveigor  and @yjshen !

**Pull request merged**